### PR TITLE
Add systems class for draw/update

### PR DESCRIPTION
**Pull request by gabija-savickaite**

### CODE REVIEW

File: source/enemies/enemy.lua [hunk 1]
```diff
@@ -54,7 +55,7 @@ function spawnEnemy(x)
 
 end
 
-function enemies:update(dt)
+function update(dt)
```
Comment (gabija-savickaite) [inline source/enemies/enemy.lua]: `local function`

File: source/systems.lua [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    systems.draw[priority] = callback
+end
+
+systemsMetatable.__index = {}
+systemsMetatable.__index.addUpdateFunction = addUpdateFunction
+systemsMetatable.__index.addDrawFunction = addDrawFunction
+
+setmetatable(systems, systemsMetatable)
+
+return systems
```
Comment (gabija-savickaite) [inline source/systems.lua:24]: `return setmetatable(systems, systemsMetatable)`

File: source/enemies/enemy.lua [hunk 3]
```diff
@@ -28,7 +28,7 @@ function spawnEnemy(x)
 
         for _, v in ipairs(towers) do
 
-            if distance(v.x, v.y, self.x, self.y) < v.radius + self.radius and self.y < v.y then
+            if util.distance(v.x, v.y, self.x, self.y) < v.radius + self.radius and self.y < v.y then
```
Comment (bradsharp) [inline source/enemies/enemy.lua:32]: `distance` is actually the magnitude of `pointA - pointB`

File: source/startup/main_require.lua [hunk 4]
```diff
@@ -1,6 +1,6 @@
-function getGlobals()
+return function()
 
-    require("source/global/utilities")
+    util = require("source/global/utilities")
     require("source/global/gameState")
     gameStateInit()
```
Comment (bradsharp) [inline source/startup/main_require.lua:4]: Should not be global

File: source/startup/startup.lua [hunk 5]
```diff
@@ -18,10 +18,10 @@ return function()
 
     -- GAME SETUP
 
-    require("source/startup/main_require")
+    local getGlobals = require("source/startup/main_require")
     getGlobals()
 
     calculatePositions()
-    populateInventory()
+    inventory.populateInventory()
```
Comment (bradsharp) [inline source/startup/startup.lua]: Could probably rename this since it starts with `inventory` now

File: source/systems.lua [hunk 6]
```diff
@@ -2,20 +2,12 @@ local systems = {}
...
 
-local systemsMetatable = {}
-
-local function addUpdateFunction(callback)
-    table.insert(systems.update, callback)
-end
-
-local function addDrawFunction(callback, priority)
+function systems.addDrawFunction(callback, priority)
     systems.draw[priority] = callback
```
Comment (bradsharp) [inline source/systems.lua]: This only allows you to have a single function at each priority with no overlap. Instead you should iterate over all draw functions and slot it in based on the priority.

File: source/towers/inventory.lua [hunk 7]
```diff
@@ -20,8 +20,8 @@ end
 function onPress(x, y)
```
Comment (bradsharp) [inline source/towers/inventory.lua]: Still global

File: source/towers/inventory.lua [hunk 8]
```diff
@@ -42,7 +42,7 @@ function onRelease(x, y)
 
         for _, v in ipairs(towerPositions) do
             if math.abs(x - v.x) < squareSide / 2 and math.abs(y - v.y) < squareSide / 2 and v.occupied == false then
-                spawnTower(v.x, v.y, itemSelected.type)
+                towers.spawnTower(v.x, v.y, itemSelected.type)
```
Comment (bradsharp) [inline source/towers/inventory.lua]: towers.spawn

File: source/towers/inventory.lua [hunk 9]
```diff
@@ -52,7 +52,13 @@ function onRelease(x, y)
 
 end
 
-function populateInventory()
+local function addToInventory(x, y, type)
+
+    table.insert(inventory, towers.getTowerModel(x, y, type))
+
```
Comment (bradsharp) [inline source/towers/inventory.lua]: y is there whitespace

**Pull request merged**